server/src/main/java/org/apache/druid/segment/loading/StorageLocation.java

```diff
@@ -117,6 +118,17 @@ public synchronized File reserve(String segmentDir, DataSegment segment)
     return reserve(segmentDir, segment.getId().toString(), segment.getSize());
   }
 
+  public synchronized boolean isReserved(String segmentDir)
+  {
+    final File segmentFile = new File(path, segmentDir);
+    return files.contains(segmentFile);
+  }
+
+  public File segmentDirectoryAsFile(String segmentDir)
+  {
+    return new File(path, segmentDir);
+  }
+
```

Comment:
The LGTM error in https://lgtm.com/projects/g/apache/druid/rev/pr-1ff2ba29372f1d2b44941bb55f75b5830f808401 seems like a false alarm. Perhaps we should suppress it for this change.